File: storm-dialect-tck/src/main/java/st/orm/tck/model/PetType.java
package st.orm.tck.model;

import lombok.Builder;
import org.jspecify.annotations.Nullable;
import st.orm.Entity;
import st.orm.PK;

@Builder(toBuilder = true)
public record PetType(
        @PK Integer id,
        String name,
        @Nullable String description
) implements Entity<Integer> {}

File: storm-ktor-koin/src/test/kotlin/st/orm/ktor/koin/model/PetType.kt
package st.orm.ktor.koin.model

import st.orm.Entity
import st.orm.GenerationStrategy
import st.orm.PK

data class PetType(
    @PK(generation = GenerationStrategy.NONE) val id: Int = 0,
    val name: String,
) : Entity<Int>
